pkg/cgroups/cgroups.go

```diff
@@ -269,12 +269,90 @@ func parseCgroupSubSysIds(filePath string) error {
 	return nil
 }
 
+// parseCgroupv2RootControllers() parses the controllers from
+// /sys/fs/cgroup/cgroup.controllers that indicates active controllers
+// that are propagated from root cgroup.
+func parseCgroupv2RootControllers() error {
+	file := filepath.Join(fmt.Sprintf("%s/1/root/%s", option.Config.ProcFS, cgroupFSPath), "cgroup.controllers")
+	data, err := os.ReadFile(file)
+	if err != nil {
+		return fmt.Errorf("failed to read %s: %v", file, err)
+	}
+
+	activeControllers := strings.TrimRight(string(data), "\n")
+	if len(activeControllers) == 0 {
+		return fmt.Errorf("no active controllers from '%s'", file)
+	}
+
+	logger.GetLogger().WithFields(logrus.Fields{
+		"cgroup.fs":          cgroupFSPath,
+		"cgroup.controllers": strings.Fields(activeControllers),
+	}).Debug("Cgroupv2 available controllers")
+
+	fixed := false
+	for i, controller := range CgroupControllers {
+		if strings.Contains(activeControllers, controller.Name) {
+			CgroupControllers[i].Id = 0
+			CgroupControllers[i].Active = true
+			fixed = true
```

Comment:
Do we need thus for v2? My understanding was that the first patch modified the bpf code to just use ->dfl_cgroup instead.

Reply:
I simplified it a bit and removed this part, and kept the environment logging logic so in case of race conditions or any other container association failure we can try to reproduce.

Reply:
> I simplified it a bit and removed this part, and kept the environment logging logic so in case of race conditions or any other container association failure we can try to reproduce.

Not sure I understand. Can you please explain?

Reply:
> > I simplified it a bit and removed this part, and kept the environment logging logic so in case of race conditions or any other container association failure we can try to reproduce.
> 
> Not sure I understand. Can you please explain?

hmm what part exactly?  also the commit log contains explanation and some extra code comments

Reply:
Discussed offline. There seem to be some races, and we have logging to detect them.